docs/source/tutorials/semantic_image_search.ipynb

```diff
@@ -0,0 +1,1222 @@
+    "    text_features[0, : len(all_tokens[0])] = torch.tensor(all_tokens)\n",
+    "\n",
+    "    # encode text\n",
+    "    embedding = clip_model._model.encode_text(text_features).to(device)\n",
+    "\n",
+    "    # convert to list for Pinecone\n",
+    "    return embedding.tolist()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d3aaabdb-e882-4790-8ac0-8007fa02b1d8",
+   "metadata": {},
+   "source": [
+    "To generate the embedding for the prompt “a picture of a giraffe”, we could do so with"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "0d2008f1-5ee6-48c3-b4bd-6e2e734e16c0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "prompt = \"a picture of a giraffe\"\n",
+    "query_vector = get_text_embedding(prompt, model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aae836d2-1392-49bb-b8fb-1eefc89a9a62",
+   "metadata": {},
+   "source": [
+    "## Creating the vector index"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b8060859-fe07-4096-8e4d-49a135fba9be",
+   "metadata": {},
+   "source": [
+    "If you have a paid Pinecone account, then you can work with multiple *vector indices* at once. However, for this article we’ll be assuming only a free Pinecone account, which limits you to one vector index. In this case, you should run the following code block to delete any existing vector index associated with your account before creating a new one."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "9128ac14-07b2-443e-ba30-32c3f0da7505",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "indices = pinecone.list_indexes()\n",
+    "if len(indices) > 0:\n",
+    "    pinecone.delete_index(indices[0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "907f019d-92a0-4bff-b487-171525d1619b",
+   "metadata": {},
+   "source": [
+    "After doing so, you can create a new vector index:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "7a2678b6-5f9c-4cc5-b25f-fa592a42a82a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "index_name = \"my-index\"\n",
+    "pinecone.create_index(index_name, dimension=512, metric=\"cosine\", pod_type=\"p1\")\n",
+    "index = pinecone.Index(index_name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5c60044b-bfdb-4b58-a387-8670f3c9efd8",
+   "metadata": {},
+   "source": [
+    "Here we have named the index “my-index” for illustrative purposes, but you can use whatever name you would like. We’ve passed in `dimension=512` because that is the dimension of the embedding vectors generated by CLIP, and we have chosen to use a “cosine” metric - vectors will be indexed according to their [cosine similarity](https://en.wikipedia.org/wiki/Cosine_similarity) to the query vector. The `pod_type=”p1”` refers to the hardware running the Pinecone service; [p1 pods](https://docs.pinecone.io/docs/indexes#p1-pods) support up to one million vectors"
```

Comment:
```suggestion
    "Here we have named the index “my-index” for illustrative purposes, but you can use whatever name you would like. We’ve passed in `dimension=512` because that is the dimension of the embedding vectors generated by CLIP, and we have chosen to use a “cosine” metric - vectors will be indexed according to their [cosine similarity](https://en.wikipedia.org/wiki/Cosine_similarity) to the query vector. The `pod_type=”p1”` refers to the hardware running the Pinecone service; [p1 pods](https://docs.pinecone.io/docs/indexes#p1-pods) support up to one million vectors."
```